frontend/src/download.js

```diff
@@ -0,0 +1,142 @@
+const UIWrapper = require('./ui').UIWrapper;
+
+let download = () => {
+  let xhr = new XMLHttpRequest();
+  xhr.open('get', '/assets' + location.pathname.slice(0, -1), true);
+  xhr.responseType = 'blob';
+
+  let listelem = setupUI();
+  xhr.addEventListener('progress', updateProgress.bind(null, listelem));
+
+  xhr.onload = function(e) {
+    // maybe send a separate request before this one to get the filename?
+
+    // maybe render the html itself with the filename, since it's generated server side
+    // after a get request with the unique id
+    listelem.emit(
+      'name',
+      xhr.getResponseHeader('Content-Disposition').match(/filename="(.+)"/)[1]
```

Comment:
Possibly. We've got a lot of error handling left to do, this included. Right now we're only thinking happy thoughts :)